### PR TITLE
Automatically add "S-waiting-on-deploy" label on a merged PR

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I used [this action](https://github.com/actions-ecosystem/action-add-labels) to add the label and [this one](https://github.com/actions-ecosystem/action-remove-labels) to remove labels. The only extra thing needed is to create a token. For which event to used, I used [github docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request).

### CODE REVIEW

File: .github/workflows/tag-merged-pr.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    branches: 
+      - master
+    types: [closed]
+
+jobs:
+  my-action:
+    if: ${{ github.event.pull_request.merged }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (Nemo157) [inline .github/workflows/tag-merged-pr.yml:12]: I'd assume it shouldn't actually be necessary to checkout for this action.

**Comment by Nemo157** [comment]
I don't think creating a token should be needed, there is an auto-generated token per GHA run that should have the required permissions (since this is running on this repo after the PR gets merged).

File: .github/workflows/tag-merged-pr.yml (hunk 1 above)
Comment (GuillaumeGomez) [reply .github/workflows/tag-merged-pr.yml]: Indeed!

**Comment by GuillaumeGomez** [comment]
I wasn't sure about that but if you think we don't need to add a token, then I guess it's ready to be tested. :)

**Review by Nemo157 — APPROVED** [review]

File: .github/workflows/tag-merged-pr.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    types: [closed]
+
+jobs:
+  my-action:
+    if: ${{ github.event.pull_request.merged }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions-ecosystem/action-add-labels@v1
+        with:
+          labels: S-waiting-on-deploy
```
Comment (jyn514) [inline .github/workflows/tag-merged-pr.yml:14]: Can we also remove all other status labels, like waiting-on-review and waiting-on-author?

Comment (GuillaumeGomez) [reply .github/workflows/tag-merged-pr.yml:14]: Sure!

**Comment by GuillaumeGomez** [comment]
I updated the action to remove the "S-waiting-*" labels as well.

**Review by syphar — APPROVED** [review]
I did  a short test of the action steps on another repo and both seem to work. 

I'm not 100% certain on the `if`, but as often with GH actions we will see when it's merged.

**Comment by syphar** [comment]
@GuillaumeGomez after merging I saw it failing. 

It seems like the default token in actions only has readonly permissions when the action is being triggered by a fork. 

I didn't dig deeper on this, in the end the people merging PRs also have write-permissions, so the action should be fine...

**Pull request merged**